File: code_examples/4-Functions/4-Functions_with_Multiple_Parameters/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses user defined functions
 */

 package main

 import (
	 "fmt"
 )
 
 func calculateArea(length int, width int) {
	 // this function calculates the area of a rectangle
 
	 // process
	 area := length * width
 
	 return area
 }
 
 func main() {
	 // this function does the input and output
	 var area = 0
 
	 // input
	 var length_from_user, width_from_user int
	 fmt.Print("Enter the length of a rectangle (cm): ")
	 fmt.Scanln(&length_from_user)
	 fmt.Print("Enter the width of a rectangle (cm): ")
	 fmt.Scanln(&width_from_user)
	 fmt.Println()
 
	 // call functions
	 area = calculateArea(length_from_user, width_from_user)
 
	 // output
	 fmt.Printf("The area is %d cm²\n", area)
 
	 fmt.Println("\nDone.")
 }
 

File: code_examples/4-Functions/4-Functions_with_Multiple_Parameters/CPP/main.cpp
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program calculates the area of a rectangle

#include <iostream>


int calculateArea(int length, int width) {
    // this function calculates the area of a rectangle

    // process
    int area = length * width;

    return area;
}

int main() {
    // this function does the input and output
    int area = 0;
    int perimeter = 0;

    // input
    std::cout << "Enter the length of a rectangle (cm): ";
    int lengthFromUser;
    std::cin >> lengthFromUser;
    std::cout << "Enter the width of a rectangle (cm): ";
    int widthFromUser;
    std::cin >> widthFromUser;
    std::cout << std::endl;

    // call functions
    area = calculateArea(lengthFromUser, widthFromUser);

    // output
    std::cout << "The area is " << area << " cm²\n";

    std::cout << "\nDone." << std::endl;
    return 0;
}
